df: extract HumanReadable from BlockSize

cakebaker opened the pull request:
This is a refactoring and consists of three parts:

1. Introduction of a `HumanReadable` enum containing the human readable variants from `BlockSize`
2. Addition of a `human_readable` field to the `Options` struct
3. Movement of header logic from `BlockSize` to `Header`

tertsdiepraam commented:
Shouldn't this be something like:
```rust
enum HumanReadable {
    Decimal,
    Binary,
}

enum BlockSize {
    Bytes(u64),
    HumanReadable(HumanReadable),
}
```
because these options should still be linked, right? I feel like this PR organizes the data in a way that is easier to misuse if that makes sense. Maybe I'm missing something, though. What was you reason for writing it this way?

cakebaker commented:
Good question, thanks.

I separated `HumanReadable` and `BlockSize` for conceptual reasons. A `BlockSize` has a visual representation (e.g. the `1K` in the `1K-blocks` header) and it defines how _byte_ values are displayed. A `HumanReadable`, on the other hand, has no visual representation and it defines how byte _and_ inode values are displayed.

A "problem" of my proposed solution is that the mutual exclusivity between `HumanReadable` and `BlockSize` is not shown. Maybe it would be helpful to introduce a third enum to resolve it? Something like:

```rust
enum OutputMode {
    HumanReadable(HumanReadable),
    BlockSize(BlockSize), // not sure about the name for this variant
}

enum HumanReadable {
    Decimal,
    Binary,
}

enum BlockSize {
    Bytes(u64),
}
```

tertsdiepraam commented:
Thank you for the explanation! Your reasoning makes a lot more sense now. It's a tricky set of options to model. It definitely needs some renaming from the current situation. I like the `OutputMode` suggestion or maybe something like `SizeFormat`. In your last example, why do you prefer the enum with a single variant?

cakebaker commented:
The latest version contains the implementation from my previous comment. I renamed `OutputMode` to `SizeFormat` to avoid confusion with the `--output` param, and replaced the `NonHumanReadable` variant with `StaticBlockSize`.

tertsdiepraam commented:
Looks great! Thanks!
